File: client/web/sudoo/lib/data/api/auth/request/account_request.dart


import 'package:sudoo/data/api/base_request.dart';

import '../../../../domain/model/auth/role.dart';

class AccountRequest extends BaseRequest {
  final String emailOrPhoneNumber;
  final String password;
  final Role role;

  AccountRequest({required this.emailOrPhoneNumber, required this.password, this.role = Role.STAFF});

  @override
  Map<String, dynamic> toJson() {
    return {
      "emailOrPhoneNumber": emailOrPhoneNumber,
      "password": password,
      "role": role.value,
    };
  }
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/data/api/auth/request/AccountRequest.kt
package com.sudo248.sudoo.data.api.auth.request

import com.google.gson.annotations.SerializedName
import com.sudo248.sudoo.domain.entity.auth.Provider


/**
 * **Created by**
 *
 * @author *Sudo248*
 * @since 09:21 - 04/03/2023
 */
data class AccountRequest(
    @SerializedName("emailOrPhoneNumber")
    val phoneNumber: String,
    val password: String,
    val provider: Provider? = null
)